Prox/Prox/Widgets/ReviewScoreView.swift

```diff
@@ -62,34 +57,58 @@ class ReviewScoreView: UIStackView {
         }
     }
 
+    private func updateReviewScore(score: Float) {
+        for (index, subview) in arrangedSubviews.enumerated() {
+            if let scoreItem = subview as? ReviewScoreItemView {
+                let scoreLeft = self.score - Float(index)
+                setFillScore(score: scoreLeft, forView: scoreItem)
+            }
+        }
+    }
+
+    private func setFillScore(score: Float, forView view: ReviewScoreItemView) {
+        if score >= 1 {
+            view.leftColor = color
+            view.rightColor = color
+        } else if score > 0 {
+            view.leftColor = color
+            view.rightColor = Colors.reviewScoreDefault
+        } else {
+            view.leftColor = Colors.reviewScoreDefault
+            view.rightColor = Colors.reviewScoreDefault
+        }
+    }
+
 }
 
 private class ReviewScoreItemView: UIView {
 
-    var color: UIColor? {
+    var leftColor: UIColor? {
         didSet {
             setNeedsDisplay()
         }
     }
-    var fill: Bool = false {
+
+    var rightColor: UIColor? {
         didSet {
-            if fill != oldValue {
-                setNeedsDisplay()
-            }
+            setNeedsDisplay()
         }
     }
 
     override func draw(_ rect: CGRect) {
         super.draw(rect)
 
-        if let context = UIGraphicsGetCurrentContext() {
-            if fill {
-                context.setFillColor(color?.cgColor ?? UIColor.black.cgColor)
-            } else {
-                context.setFillColor(Colors.reviewScoreDefault.cgColor)
-            }
-            context.fillEllipse(in: rect)
-        }
+        let width = rect.size.width / 2
+
+        let leftHalf = UIBezierPath(roundedRect: CGRect(x: rect.origin.x, y: rect.origin.x, width: width, height: rect.size.height), byRoundingCorners: [UIRectCorner.topLeft, UIRectCorner.bottomLeft], cornerRadii: CGSize(width: 10, height: 10))
+        leftHalf.close()
+        (leftColor ?? UIColor.black).setFill()
```

Comment:
Why `UIColor.black`? We previously set `Colors.reviewScoreDefault` – should we make that the default? Is it for uninitialized errors? In which case I'd recommend throwing.